golangci-lint does not build on darwin-x86_64 - undefined symbols

golangci-lint on unstable does not build on darwin-x86_64. This seems to have been caused by #166801 which switched the go version from 1.17 to 1.18 (to fix another (runtime) bug!)

See hydra: https://hydra.nixos.org/job/nixpkgs/trunk/golangci-lint.x86_64-darwin

```
/nix/store/gynxiihx8rd89nid1ma8pmj9b3nd3kzz-go-1.18.1/share/go/pkg/tool/darwin_amd64/link: running clang failed: exit status 1
Undefined symbols for architecture x86_64:
  "_SecTrustEvaluateWithError", referenced from:
      _crypto/x509/internal/macos.x509_SecTrustEvaluateWithError_trampoline.abi0 in go.o
  "_utimensat", referenced from:
      _syscall.libc_utimensat_trampoline.abi0 in go.o
ld: symbol(s) not found for architecture x86_64
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
```

FYI (sorry to ping but not really sure who else I would ask)
@Arriven @siraben - author/merger of #166801
@Mic92 - arbitrary maintainer


```console
❯ nix run nixpkgs#nix-info -- -m
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.5.1`
 - channels(root): `"nixpkgs-22.05pre344086.568e0bc498e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

--- a/pkgs/applications/misc/hugo/default.nix
+++ b/pkgs/applications/misc/hugo/default.nix
@@ -33,7 +33,6 @@ buildGoModule rec {
   '';
 
   meta = with lib; {
-    broken = stdenv.isDarwin && stdenv.isx86_64;
     description = "A fast and modern static website engine";
     homepage = "https://gohugo.io";
     license = licenses.asl20;

--- a/pkgs/development/tools/golangci-lint/default.nix
+++ b/pkgs/development/tools/golangci-lint/default.nix
@@ -35,6 +35,5 @@ buildGoModule rec {
     homepage = "https://golangci-lint.run/";
     license = licenses.gpl3Plus;
     maintainers = with maintainers; [ anpryl manveru mic92 ];
-    broken = stdenv.isDarwin && stdenv.isx86_64;
   };
 }

